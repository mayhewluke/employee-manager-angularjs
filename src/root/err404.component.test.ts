import angular, { ICompileService, IScope } from "angular";
import "angular-mocks";

import err404Component from "./err404.component";

describe("err404 component", () => {
  let render: () => JQLite;
  let scope: IScope;
  beforeEach(angular.mock.module(err404Component));
  beforeEach(
    angular.mock.inject(($rootScope: IScope, $compile: ICompileService) => {
      scope = $rootScope.$new();
      render = () => $compile("<err404></err404>")(scope);
    }),
  );

  it("displays an error message", () => {
    const element = render();

    expect(element.find(".error")[0].outerHTML).toMatchSnapshot();
  });

  it("provides a link back to the app", () => {
    const element = render();

    expect(element.find("a[ui-sref]").attr("ui-sref")).toEqual("init");
  });
});
