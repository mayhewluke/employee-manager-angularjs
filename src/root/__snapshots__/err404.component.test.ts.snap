// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`err404 component displays an error message 1`] = `"<h1 class=\\"error\\">Error 404: Not Found</h1>"`;
